aiida/backends/sqlalchemy/models/group.py

```diff
@@ -26,7 +26,10 @@
     Base.metadata,
     Column('id', Integer, primary_key=True),
     Column('dbnode_id', Integer, ForeignKey('db_dbnode.id', deferrable=True, initially="DEFERRED")),
-    Column('dbgroup_id', Integer, ForeignKey('db_dbgroup.id', deferrable=True, initially="DEFERRED"))
+    Column('dbgroup_id', Integer, ForeignKey('db_dbgroup.id', deferrable=True, initially="DEFERRED")),
+
+    # explicit/composite unique constraint.  'name' is optional.
+    UniqueConstraint('dbnode_id', 'dbgroup_id', name='uix_dbnode_id_dbgroup_id')
```

Comment:
Is adding this constraint necessary or is it just to be more in line with aiida 1.0?

Reply:
It is needed for the efficient duplicate detection. By adding the constraint we pass this check to the database (please look at the SQL query but also the SQLA non-ORM code)  and this is the most efficient, in terms of speed, thing to do.